src/Cache/SysVCacheItemPool.php

```diff
@@ -0,0 +1,231 @@
+<?php
+/**
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+namespace Google\Auth\Cache;
+
+use Psr\Cache\CacheItemInterface;
+use Psr\Cache\CacheItemPoolInterface;
+
+/**
+ * SystemV shared memory based CacheItemPool implementation.
+ *
+ * This CacheItemPool implementation can be used among multiple processes, but
+ * it doesn't provide any locking mechanism. If multiple processes write to
+ * this ItemPool, you have to avoid race condition manually in your code.
+ */
+class SysVCacheItemPool implements CacheItemPoolInterface
+{
+    const VAR_KEY = 1;
+
+    const DEFAULT_PROJ = 'A';
+
+    const DEFAULT_MEMSIZE = 10000;
+
+    const DEFAULT_PERM = 0600;
+
+    /** @var int */
+    private $sysvKey;
+
+    /**
+     * @var CacheItemInterface[]
+     */
+    private $items;
+
+    /**
+     * @var CacheItemInterface[]
+     */
+    private $deferredItems;
+
+    /**
+     * @var array
+     */
+    private $options;
+
+    /**
+     * Save the current items.
+     *
+     * @return bool true when success, false upon failure
+     */
+    private function saveCurrentItems()
```

Comment:
A good practice is to following the following flow of method defintions:

- public
- protected
- private